pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/nullvalue/NullValueVectorCreator.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.creator.impl.nullvalue;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.pinot.core.segment.creator.impl.V1Constants;
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+import java.io.Closeable;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * Used to persist the null bitmap on disk. This is used by SegmentCreator
+ * while indexing rows.
+ */
+public class NullValueVectorCreator implements Closeable {
+
+  private MutableRoaringBitmap _nullBitmap;
+  private File _nullValueVectorFile;
+
+  public NullValueVectorCreator(File indexDir, String columnName) {
+    _nullValueVectorFile = new File(indexDir, columnName + V1Constants.Indexes.NULLVALUE_VECTOR_FILE_EXTENSION);
+    _nullBitmap = new MutableRoaringBitmap();
+  }
+
+  @Override
+  public void close()
```

Comment:
Recommend having a `seal()` method similar to `InvertedIndexCreator` instead of using `close()` to persist the data.

Reply:
Any particular reason why ? It's using the same approach as Bloom Filter creator

Reply:
The semantic for `close()` is "Closes this resource, relinquishing any underlying resources." and it should not be responsible for persisting the data. Bloom Filter should also follow this, but that is out of the scope of this PR.

Reply:
done -> added seal